datafusion/core/src/datasource/schema_adapter.rs

```diff
@@ -75,9 +75,16 @@ pub trait SchemaAdapter: Send + Sync {
 
 /// Creates a `SchemaMapping` that can be used to cast or map the columns
 /// from the file schema to the table schema.
-pub trait SchemaMapper: Send + Sync {
+pub trait SchemaMapper: Debug + Send + Sync {
     /// Adapts a `RecordBatch` to match the `table_schema` using the stored mapping and conversions.
     fn map_batch(&self, batch: RecordBatch) -> datafusion_common::Result<RecordBatch>;
+
+    /// Adapts a `RecordBatch` that does not have all the columns (as defined in the schema).
```

Comment:
I don't fully understand the need for this function (maybe it would be better to modify the schema `file_schema` to have the columns that are needed 🤔

Reply:
As I understand it, when `DatafusionArrowPredicate::evaluate` is called, the `RecordBatch` only contains one column. If we use the `map_batch` function, it indexes into the `RecordBatch` as if all the columns are there.

For example, if we have a schema with fields: `[{name: "value", type: "Float64"},{name: "time", type: "Timestamp"}]`, then map_batch will try to index at `1` for time, but the `RecordBatch` won't actually have that index since only one column is passed in.

This definitely may be the wrong way to achieve what I wanted, but the intent was to look the field up by name in the `table_schema` so that we don't have that indexing problem.